addons/nightvision/functions/fnc_setupDisplayEffects.sqf

```diff
@@ -0,0 +1,78 @@
+/*
+ * Author: Dslyecxi, PabstMirror
+ * Handles setting up the effects: fog, ppEffects and the RscTittle
+ *
+ * Arguments:
+ * 0: Activated <BOOL>
+ *
+ * Return Value:
+ * Nothing
+ *
+ * Example:
+ * [true] call ace_nightvision_fnc_setupDisplayEffects
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_activated"];
+TRACE_1("setup",_activated);
+
+// Backup and restore changes to fog:
+// Note: modifying the fog
+if (missionNamespace getVariable [QGVAR(enableChangingFog), true]) then {
+    if (_activated) then {
+        if (isNil QGVAR(priorFog)) then {
+            TRACE_1("Backing up fog",fogParams);
+            GVAR(priorFog) = fogParams;
+        } else {
+            LOG("fog already backed up");
+        };
+
+        // Handle non-dedicated:
+        if (isServer) then {
+            missionNamespace setVariable [QGVAR(serverPriorFog), fogParams, true];
+            [QGVAR(nonDedicatedFix), []] call CBA_fnc_globalEvent;
+        };
+    } else {
+        if (!isNil QGVAR(priorFog)) then {
+            0 setFog GVAR(priorFog);
+            GVAR(priorFog) = nil;
+        } else {
+            ERROR("no fog backed up");
+        };
+
+        // Handle non-dedicated:
+        if (isServer) then {
+            missionNamespace setVariable [QGVAR(serverPriorFog), nil, true];
+        };
+    };
+};
+
+// Note: Using BIS_fnc_rscLayer because of bug with string syntax - https://feedback.bistudio.com/T120768
+(QGVAR(display) call BIS_fnc_rscLayer) cutText ["", "PLAIN"]; // Cleanup Old Display
+if (_activated) then { // Create New Display
+    (QGVAR(display) call BIS_fnc_rscLayer) cutRsc [QGVAR(title), "PLAIN", 0, false];
```

Comment:
Wouldn't the game focus on the created control? Displaying a cursor and leaving the player unable to perform any actions?